instrumentation/opentelemetry-instrumentation-aio-pika/tests/test_publish_decorator.py

```diff
@@ -66,7 +69,60 @@ def test_get_publish_span(self):
         )
 
     def _test_publish(self, exchange_type: Type[Exchange]):
-        exchange = exchange_type(CONNECTION, CHANNEL, EXCHANGE_NAME)
+        exchange = exchange_type(CONNECTION_7, CHANNEL_7, EXCHANGE_NAME)
+        with mock.patch.object(
+            PublishDecorator, "_get_publish_span"
+        ) as mock_get_publish_span:
+            with mock.patch.object(
+                Exchange, "publish", return_value=asyncio.sleep(0)
+            ) as mock_publish:
+                decorated_publish = PublishDecorator(
+                    self.tracer, exchange
+                ).decorate(mock_publish)
+                self.loop.run_until_complete(
+                    decorated_publish(MESSAGE, ROUTING_KEY)
+                )
+        mock_publish.assert_called_once()
+        mock_get_publish_span.assert_called_once()
+
+    def test_publish(self):
+        self._test_publish(Exchange)
+
+    def test_robust_publish(self):
+        self._test_publish(RobustExchange)
+
+
+@skipIf(version_info <= (8, 0), "Only for aio_pika 8")
+class TestInstrumentedExchangeAioRmq8(TestCase):
+    EXPECTED_ATTRIBUTES = {
+        SpanAttributes.MESSAGING_SYSTEM: MESSAGING_SYSTEM,
+        SpanAttributes.MESSAGING_DESTINATION: f"{EXCHANGE_NAME},{ROUTING_KEY}",
+        SpanAttributes.NET_PEER_NAME: SERVER_HOST,
+        SpanAttributes.NET_PEER_PORT: SERVER_PORT,
+        SpanAttributes.MESSAGING_MESSAGE_ID: MESSAGE_ID,
+        SpanAttributes.MESSAGING_CONVERSATION_ID: CORRELATION_ID,
+        SpanAttributes.MESSAGING_TEMP_DESTINATION: True,
+    }
+
+    def setUp(self):
+        self.tracer = get_tracer(__name__)
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+
+    def test_get_publish_span(self):
+        exchange = Exchange(CHANNEL_8, EXCHANGE_NAME)
```

Comment:
This `Exchange` constructor was a change between 7 and 8, and this is also what prompted me to decorate with `skipIf`.